contribs/src/main/scala/io/delta/storage/DynamoDBLogStore.scala

```diff
@@ -0,0 +1,185 @@
+  - spark.delta.DynamoDBLogStore.region - AWS region (defaults to 'us-east-1')
+  - spark.delta.DynamoDBLogStore.fakeAuth - use for dynamodb-local (defaults to 'false')
+
+*/
+class DynamoDBLogStore(
+  sparkConf: SparkConf,
+  hadoopConf: Configuration) extends BaseExternalLogStore(sparkConf, hadoopConf) {
+
+  import DynamoDBLogStore._
+
+  private val tableName = sparkConf.get(s"${confPrefix}tableName", "delta_log")
+
+  private val client: AmazonDynamoDBClient = DynamoDBLogStore.getClient(sparkConf)
+
+  override protected def writeCache(
+    fs: FileSystem,
+    logEntry: LogEntryMetadata,
+    overwrite: Boolean): Unit = {
+    try {
+      logInfo(s"putItem $logEntry, overwrite: $overwrite")
+      client.putItem(logEntry.asPutItemRequest(tableName, overwrite))
+    } catch {
+      case e: ConditionalCheckFailedException =>
+        logError(e.toString)
+        throw new java.nio.file.FileAlreadyExistsException(logEntry.path.toString)
+      case e: Throwable =>
+        logError(e.toString)
+        throw new java.nio.file.FileSystemException(logEntry.path.toString)
+    }
+  }
+
+  override protected def listFromCache(
+    fs: FileSystem, resolvedPath: Path): Iterator[LogEntryMetadata] = {
+    val filename = resolvedPath.getName
+    val parentPath = resolvedPath.getParent
+    logInfo(s"query parentPath = $parentPath AND filename >= $filename")
+
+    val result = client
+      .query(
+        new QueryRequest(tableName)
+          .withConsistentRead(true)
+          .withKeyConditions(
+            Map(
+              "filename" -> new Condition()
+                .withComparisonOperator(ComparisonOperator.GE)
+                .withAttributeValueList(new AttributeValue(filename)),
+              "parentPath" -> new Condition()
+                .withComparisonOperator(ComparisonOperator.EQ)
+                .withAttributeValueList(new AttributeValue(parentPath.toString))
+            ).asJava
+          )
+      )
+      .getItems
+      .asScala
+
+    result.iterator.map(item => {
+      logInfo(s"query result item: ${item.toString}")
+      val parentPath = item.get("parentPath").getS
+      val filename = item.get("filename").getS
+      val tempPath = Option(item.get("tempPath").getS).map(new Path(_))
+      val length = item.get("length").getN.toLong
+      val modificationTime = item.get("modificationTime").getN.toLong
+      val isComplete = item.get("isComplete").getS() == "true"
+      LogEntryMetadata(
+        path = new Path(s"$parentPath/$filename"),
+        tempPath = tempPath,
+        length = length,
+        isComplete = isComplete,
+        modificationTime = modificationTime
+      )
+    })
+  }
+}
+
+object DynamoDBLogStore {
+  private val confPrefix = "spark.delta.DynamoDBLogStore."
+
+  implicit def logEntryToWrapper(entry: LogEntryMetadata): LogEntryWrapper = LogEntryWrapper(entry)
+
+  def getClient(sparkConf: SparkConf): AmazonDynamoDBClient = {
+
+    var client = new AmazonDynamoDBClient()
```

Comment:
Can you enable support for custom Auth class here? Something along the line of the Hadoop credentials provider

```
spark.delta.DynamoDBLogStore.credentials.provider=<credentials provider class name>
```
and the config points to an implementation of `com.amazonaws.auth.AWSCredentialsProvider` with a no-args constructor? Otherwise this would block a lot of complex auth flows

Reply:
good idea, will do.

Reply:
@andyndang the other option is configuring `AmazonDynamoDBClient` this way (pointing to no-args constructor of `AmazonDynamoDBClient`). It may be more flexible, what do you think?

Reply:
I'm not sure I follow you. Do you mean the other option is the existing code path? The default no-args client relies on env/AWS credentials file and is not very flexible. For example, it wouldn't work with AWS SSO

Reply:
Ok, so let me explain. Alternative to configuring AWSCredentialProvider (as you proposed) may be configuring AmazonDynamoDBClient the similar way: 
```spark.delta.DynamoDBLogStore.dynamodb_client =<dynamodb client no-args constructor>```
This way full customization of DynamoDBClient would be possible (including custom credential provider)

Reply:
It's extremely hard to extend existing AWS Client (the interface is non trivial), whereas the CredentialsProvider interface is trivial to implement. So IMO this option doesn't make sense.

You can technically have:
```
spark.delta.DynamoDBLogStore.dynamo_client.supplier=<a supplier for DDB client>
```
But this means you'll need to introduce a new interface + added documentation. The AWS Credentials set up is common for Spark/Hadoop users so it's a lot easier to explain to people

Reply:
done